src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/ManagedBeanQuickFix.java

```diff
@@ -48,29 +55,38 @@ protected void insertAnnotations(Diagnostic diagnostic, JavaCodeActionContext co
         }
     }
 
-    private static void insertAndReplaceAnnotation(Diagnostic diagnostic, JavaCodeActionContext context,
-                                                   List<CodeAction> codeActions, String annotation) {
-        // Diagnostic is reported on the variable declaration, however the
-        // annotations that need to be replaced are on the type declaration (class
-        // definition) containing the variable declaration. We retrieve the type
-        // declaration container here.
-        PsiElement parentNode = context.getCoveredNode();
-        PsiClass classBinding = PsiTreeUtil.getParentOfType(parentNode, PsiClass.class);
-
-        // Insert the annotation and the proper import by using JDT Core Manipulation
-        // API
+    private void insertAndReplaceAnnotation(Diagnostic diagnostic, JavaCodeActionContext context,
+                                            List<CodeAction> codeActions, String annotation) {
         String name = getLabel(annotation);
+        codeActions.add(JDTUtils.createCodeAction(context, diagnostic, name, getParticipantId()));
+    }
+
+    @Override
+    public CodeAction resolveCodeAction(JavaCodeActionResolveContext context) {
+        final CodeAction toResolve = context.getUnresolved();
+        String name = getLabel(ADD_ANNOTATION);
```

Comment:
Should retrieve the title from the code action instead of recomputing it here.

Reply:
Made the necessary changes so that the title is retrieved from `toResolve.getTitle()` .